lib/src/main/scala/com/worksap/nlp/uzushio/lib/filters/NoContentDOM.scala

```diff
@@ -1,26 +1,90 @@
 package com.worksap.nlp.uzushio.lib.filters
 
-import com.worksap.nlp.uzushio.lib.cleaning.Paragraph
+import com.worksap.nlp.uzushio.lib.cleaning.{Paragraph, PathSegment}
 import com.worksap.nlp.uzushio.lib.filters.base.ParagraphFilter
 
 class NoContentDOM extends ParagraphFilter {
-  // This names are tag names, but also class names and id names
   final private val filteringDomNames: Seq[String] =
-    Array("header", "footer", "aside", "nav", "left-box", "sidebar")
+    Array("header", "footer", "aside", "nav", "noscript", "form")
+
+  final private val DOMCandidatesForFiliteringClassOrId = Array("div", "p", "ul", "h1")
+
+  final private val filteringFullMatchClassOrIdCandidates: Seq[String] = Array(
+    "left-box",
+    "blog-title-inner",
+    "globalheader",
+    "blogtitle",
+    "blog-name",
+    "head-block1",
+    "head-blog-name",
+    "head-introduction",
+  )
+
+  final private val filteringPartialMatchClassOrIdNames: Seq[String] = Array(
+    "header",
+    "footer",
+    "side",
+    "aside",
+    "sidebar",
+    "menu",
+    "nav",
+    "navi",
+    "navigation",
+    "navbar",
+    "banner",
+    "logo",
+    "pankuzu",
+    "breadcrumb",
+    "breadcrumbs",
+    "widget",
+    // "profile",
+    "button",
+  )
+
+  final private val filteringFullMatchClassOrIdNames =
+    filteringPartialMatchClassOrIdNames ++ filteringFullMatchClassOrIdCandidates ++ filteringFullMatchClassOrIdCandidates
+      .map(toCamelCase)
+
+  def toCamelCase(s: String): String = {
+    val words = s.split("-")
+    words.head + words.tail.map(_.capitalize).mkString
+  }
+
+  def partialMatchIds(css: PathSegment): Boolean = filteringPartialMatchClassOrIdNames
+    .exists(name =>
+      css.id != null && (css.id.split("[_-]").contains(name) || css.id.capitalize
+        .contains(name.capitalize))
+    )
+
+  def partialMatchClasses(css: PathSegment): Boolean = filteringPartialMatchClassOrIdNames
+    .exists(name =>
+      css.classes.exists(_.split("[_-]").contains(name)) || css.classes
+        .exists(_.capitalize.contains(name.capitalize))
+    )
 
-  // I checked some of the Common Crawl extracts and noticed that `div#header` and `div.nav` are also often used instead of `<header>` and `<nav>`.
   def containsTagWithIdAndClasses(
       p: Paragraph,
-      tagName: String,
-      classOrIdNames: Seq[String]
+      tagNames: Seq[String],
+      fullMatchCandidates: Seq[String],
+      partialMatchCandidates: Seq[String]
   ): Boolean = {
     val iter = p.cssPath.reverseIterator
 
     while (iter.hasNext) {
       val css = iter.next()
+
+      if (
+        fullMatchCandidates
+          .exists(name => tagNames.contains(css.tag) && (css.id == name || css.classes.contains(name)))
+      ) {
+        return true
+      }
+
+      // checking filtering keywords in snake case and camel case, kebab case
```

Comment:
If matching a keyword in full texts, it may cause a False Positive in some keywords (ex. `nav`, `side`).

For example, `side` matches `seaside`, and `nav` matches `navy`.
So, I implemented matching keywords to split words.